### ISSUE TITLE
Bug: [@typescript-eslint/no-deprecated] incorrectly reports MUI `anchor` prop as deprecated

### ISSUE THREAD

**Issue by MorrisonCole** [typescript-eslint/typescript-eslint#9965]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.4&fileType=.tsx&code=JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwCwAUI8AHYxZQF5ZwBCArgHMACsTDoA3ozgoWuABbQAXOQIQItRgF9GjXBBap4AwXAC8mKjAB0aqAHdkUACbYCAHgASAFQCyAGQARYAA3AFEAGywQLDYAGj4hUQhxAD4ACgJ%2BORhgA0TBdLAxVATKAgBKOCkGGX1DeBqZGWQ5RShzcgAjJzJpOC1O4pTUeiZauAB6SbhsygjkdmcEgCt%2BI1nUHjs4VBIsGHlWQX76vajrCIhC8oqxmUoYfigWOHdUiVaFaC13SdSxlo7roGGcIBcroUTHcgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import * as React from 'react';

interface BugProps {
  anchor: 'foo';
}

const Bug = React.forwardRef<HTMLDivElement, BugProps>(function Bug(props, ref) {
  const {
    anchor = 'bar'
  } = props;

  return <></>;
});
```
```


### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-deprecated": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error reported by `"@typescript-eslint/no-deprecated": "error"`, since `anchor` is my own property and it is not deprecated.

### Actual Result

`anchor` is reported as deprecated.

### Additional Info

I've checked that the react prop forwarding doesn't shadow the `anchor` property:

```
import * as React from 'react';

interface BugProps {
  myAnchor: 'foo';
}

const Bug = React.forwardRef<HTMLDivElement, BugProps>(function Bug(props, ref) {
  const {
    anchor = 'bar'
  } = props;

  return <></>;
});
```
With this, the only error is from TS - `Property 'anchor' does not exist on type 'BugProps'.`.

Also, the lint rule doesn't report the error if I use direct assignment rather than props destructuring:

```
const noProblem = props.anchor;
```

**Comment by bradzacher** [comment]
The repro can be simplified to just
```
interface BugProps {
  anchor: 'foo';
}
declare const x: BugProps;
const { anchor = "" } = x;
```

Note that `const { anchor } = x` does not report, nor does `const { anchor = 1 } = x` -- it specifically has to be given with a string default

**Comment by auvred** [comment]
```ts
interface Props {
  /** @deprecated */
  foo: Props
}
declare const x: Props;
const { foo = x } = x;
```

`foo` is reported twice here, so this is not just an issue with global types (deprecated `anchor` of `String` interface is just a coincidence)
